src/kudu/consensus/consensus_queue.cc

```diff
@@ -1041,23 +1041,18 @@ void PeerMessageQueue::GetDataCommitQuorum(
       &voter_distribution);
 
   // Step 2: Compute data commit quorum from the config.
-  for (const RegionRuleSpecPB& rule_spec :
-      queue_state_.active_config->commit_rule().rule_conjunctions()) {
-    std::string region = rule_spec.region_identifier();
-    const std::string& commit_req = rule_spec.commit_requirement();
-
-    // Resolve master region.
-    if (region == "master") {
-      CHECK(local_peer_pb_.has_attrs() && local_peer_pb_.attrs().has_region());
-      region = local_peer_pb_.attrs().region();
-    }
-
-    // TODO(ritwikyadav): It is possible that all voters in a region
-    // have left the replicaset. Handle that case gracefully in a subsequent diff.
-    int total_voters = FindOrDie(voter_distribution, region);
-    int commit_req_num = ResolveCommitRequirement(total_voters, commit_req);
-    InsertIfNotPresent(data_commit_quorum, region, commit_req_num);
-  }
+  // TODO(ritwikyadav) : This function will be simplified (or removed) shortly
+  // with the new quorum specification modes.
+  CHECK(queue_state_.active_config->commit_rule().mode() ==
+      QuorumMode::SINGLE_REGION_DYNAMIC);
+
+  // Double check that the local leader has the proper metadata.
+  CHECK(local_peer_pb_.has_attrs() && local_peer_pb_.attrs().has_region());
+
+  const std::string& leader_region = local_peer_pb_.attrs().region();
+  int total_voters = FindOrDie(voter_distribution, leader_region);
+  int commit_req_num = MajoritySize(total_voters);
+  InsertIfNotPresent(data_commit_quorum, leader_region, commit_req_num);
```

Comment:
this logic might not function correctly in production.
Lets say we have 3 regions with 5 members each. One of the regions unfortunately has 2 instances less
than desired due to maintenance. So we have 3, 5, 5 members in R1, R2, R3.
Doing majority based on active config wont work because it would mean a quorum size of 2, 3, 3
as the primary moves. What we want is 3 votes all the time.

Reply:
Let's discuss this more tomorrow since this ties in to our discussion from today. There are multiple options here:

1 . For scheduled maintenance, we would get the replacement first and then remove the host from config. I have assumed this was the case but if it can't be achieved, there is another option on the table.

2. Instead of me converting the list of servers from the config to a voter distribution, I can accept a parameter with the commit rule which essentially gives me a mapping <region->server_count>. I would expect this to remain the same for "replace" operations like in the case of maintenance.

Reply:
This was discussed in yesterday's meeting. The resolution was that we would explicitly call out the configured server distribution which would remain untouched by automation when they are doing maintenance.